File: android/app/src/main/java/com/ptokenssentinelandroidapp/database/DatabaseInterface.java
package com.ptokenssentinelandroidapp.database;

// NOTE: This interface is use by the rust-lib this java program interacts with.
@SuppressWarnings("unused")
public interface DatabaseInterface {
    byte[] get(byte[] key, byte dataSensitivity) throws Exception;
    void put(byte[] key, byte[] value, byte dataSensitivity);
    void delete(byte[] key);
    String startTransaction();
    String endTransaction();
    void close();
}

File: android/app/src/main/java/com/ptokenssentinelandroidapp/database/DatabaseInterface.kt
package com.ptokenssentinelandroidapp

interface DatabaseInterface {
  fun close()

  fun delete(key: ByteArray?)

  fun put(key: ByteArray?, value: ByteArray?, dataSensitivity: Byte)

  @Throws(DatabaseException::class)
  fun endTransaction()

  @Throws(DatabaseException::class)
  fun startTransaction()

  @Throws(Exception::class)
  operator fun get(key: ByteArray?, dataSensitivity: Byte): ByteArray?
}